transformer/mappings.py

```diff
@@ -207,28 +217,58 @@ def __init__(self, *args, **kwargs):
         self.aspace_helper = ArchivesSpaceHelper()
         return super(ArchivesSpaceArchivalObjectToCollection, self).__init__(*args, **kwargs)
 
+    @odin.map_field
+    def title(self, value):
+        return value if value else self.source.display_string
+
+    @odin.map_field(from_field='language', to_field='languages', to_list=True)
+    def languages(self, value):
+        value = value if value else self.aspace_helper.closest_parent_value(self.source.uri, 'language')
+        lang_data = languages.get(part2b=value)
+        return [Language(expression=lang_data.name, identifier=value)]
+
     @odin.map_list_field(from_field='subjects', to_field='terms')
     def terms(self, value):
         return ArchivesSpaceRefToReference.apply(value)
 
-    @odin.map_list_field(from_field='linked_agents', to_field='agents')
-    def agents(self, value):
-        return ArchivesSpaceLinkedAgentToReference.apply(value)
-
     @odin.map_list_field(from_field='dates', to_field='dates')
     def dates(self, value):
+        if not value:
+            value = [json_codec.loads(json.dumps(d), ArchivesSpaceDate) for d in self.aspace_helper.closest_parent_value(self.source.uri, 'dates')]
```

Comment:
I am wondering if we want to return a string from `closest_parent_value()` so we don't have to do `json.dumps()` each time we use it? Are there instances in which we *don't* dump to a string in a mapping when returning from this function?

Reply:
Oh wait, that won't work because we won't be able to iterate over the return as a list if it's a string...nevermind.